src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -136,9 +140,13 @@ class ClusterMonitorActor(val cluster: Cluster,
     // Only do this if the cluster was not created with the pet service account.
     val bucketACLFuture = setStagingBucketACLsForUser()
 
+    implicit val timeout: Timeout = Timeout(5 seconds)
+    val enableClusterProxying: Future[Any] = clusterDnsCache ? ProcessReadyCluster(cluster)
```

Comment:
Since the `ClusterDnsCache.processReadyCluster` call returns a `Future`, I think the shape of `enableClusterProxying` is actually a `Future[Future[Any]]` which is probably not what we want. Probably can make `processReadyCluster` run in the actor thread and return an Either instead.